stacks-node/src/tests/signer/mod.rs

```diff
@@ -1241,48 +1241,46 @@ impl<Z: SpawnedSignerTrait> SignerTest<Z> {
         panic!("Timed out while waiting for confirmation of block with signer sighash = {block_signer_sighash}")
     }
 
-    fn wait_for_validate_ok_response(&self, timeout: Duration) -> BlockValidateOk {
+    fn wait_for_validate_ok_response(&self, timeout_secs: u64) -> BlockValidateOk {
         // Wait for the block to show up in the test observer
-        let t_start = Instant::now();
-        loop {
+        let mut validate = None;
+        wait_for(timeout_secs, || {
```

Comment:
This makes me want to add a new `wait_for` that takes a function that returns a `Result<Option<T>, String>`. We can do that in a separate PR though.